virtualmouse/fakeinput.go

```diff
@@ -0,0 +1,11 @@
+package main
+
+import (
+	"github.com/BurntSushi/xgb"
+	"github.com/BurntSushi/xgb/xproto"
+	"github.com/BurntSushi/xgb/xtest"
+)
+
+func fakeinput (X *xgb.Conn, wid xproto.Window, x int16, y int16) {
```

Comment:
Move the connection logic into your library.

Specifically, your library should provide one function that takes in the display argument and returns a struct with the connection, xproto, setup etc.

Rest of the functions should be defined on the struct itself. Stuff like movemouse (which should only need to take x, and y), or click (which should take no arguments), or getposition (which should return x, y).